src/worker/tasks/sentTxWorker.ts

```diff
@@ -0,0 +1,253 @@
+        minedAt,
+        minedAtBlock: transactionReceipt.blockNumber,
+        onChainTxStatus: transactionReceipt.status,
+        transactionHash: transactionReceipt.transactionHash,
+        transactionType: transactionReceipt.type === "legacy" ? 0 : 1, // @TODO: Fix this
+        gasPrice: transactionReceipt.effectiveGasPrice,
+        gas: transactionReceipt.gasUsed,
+        maxFeePerGas: transactionResponse.maxFeePerGas,
+        maxPriorityFeePerGas: transactionResponse.maxPriorityFeePerGas,
+        nonce: transactionResponse.nonce,
+      },
+    });
+
+    // @TODO: report usage
+  };
+
+  handleRetry = async (sentTransaction: SentTransaction) => {
+    const chain = defineChain(sentTransaction.chainId);
+    const rpcRequest = getRpcClient({ client: thirdwebClient, chain });
+    const config = await getConfig();
+
+    // Retry recently submitted transaction later.
+    const block = await eth_blockNumber(rpcRequest);
+    const ellapsedBlocks = block - sentTransaction.sentAtBlock;
+    if (ellapsedBlocks < config.minEllapsedBlocksBeforeRetry) {
+      throw new Error("Transaction is recently submitted. Trying again later.");
+    }
+
+    // Cancel unconfirmed transactions after some duration.
+    const ellapsedSeconds =
+      (Date.now() - sentTransaction.sentAt.getTime()) / 1_000;
+    if (ellapsedSeconds > this.CANCEL_TIMEOUT_SECONDS) {
+      return this.handleCancel(sentTransaction);
+    }
+
+    // Assert queuedTransaction is valid.
+    // @TODO: typing?
+    if (!sentTransaction.fromAddress) {
+      throw new Error("Queued transaction missing required fields.");
+    }
+
+    const wallet = await getWallet({
+      chainId: chain.id,
+      walletAddress: sentTransaction.fromAddress,
+    });
+    const account = await ethers5Adapter.signer.fromEthers({
+      signer: wallet.getSigner(),
+    });
+
+    // Prepare the transaction with aggressive gas settings.
+    const preparedTransaction = prepareTransaction({
+      client: thirdwebClient,
+      chain,
+      to: sentTransaction.toAddress,
+      data: sentTransaction.data as `0x${string}`,
+      value: sentTransaction.value,
+      nonce: sentTransaction.nonce,
+    });
+    const retryTransaction = await this.updateTransactionGasOptions({
+      preparedTransaction,
+      sentTransaction,
+    });
+
+    // Send transaction to RPC.
+    let transactionHash: `0x${string}`;
+    try {
+      const result = await sendTransaction({
+        transaction: retryTransaction,
+        account,
+      });
+      transactionHash = result.transactionHash;
+    } catch (e) {
+      // RPC returned an error.
+      // @TODO: flag tx as errored.
+      return;
+    }
+
+    // Update the transactions DB.
+    await updateTx({
+      queueId: sentTransaction.id,
+      data: {
+        status: TransactionStatus.Sent,
+        sentAt: new Date(),
+        sentAtBlock: await eth_blockNumber(rpcRequest),
+        transactionHash,
+        retryCount: 1,
```

Comment:
we should do `dbData.retryCount + 1` or update `updateTx()` to increment on `Retry` if not here.